src/Integrations/POSCatalog/ProductMapper.php

```diff
@@ -52,7 +77,12 @@ public function map_product( WC_Product $product ): array {
 		 * @param array      $row     Mapped product data.
 		 * @param WC_Product $product Product object.
 		 */
-		return apply_filters( 'oapfw_map_catalog_product', $row, $product );
+		$row = apply_filters( 'oapfw_map_catalog_product', $row, $product );
+
+		if ( null === $this->fields ) {
+			return $row;
+		}
+		return _rest_array_intersect_key_recursive( $row, $this->fields );
```

Comment:
Do we need recursion here? From the issue description, it seems that `array_intersect_key` is quite enough. 

```suggestion
		return array_intersect_key( $row, $this->fields );
```

Reply:
The current implementation is not what it should be, see the initial comments in PAY-430. I'll actually need to borrow a bit more from `rest_filter_response_fields()` to support stuff like `image.src`. I'll update this comment thread when I do.

Reply:
I added https://github.com/woocommerce/woocommerce-product-feed-for-openai/pull/36/commits/179a9150e5d3dad13589e9ec67cff0079e531689 to achieve the desired effect where sub-fields can be specified.

I'll ask in the Linear issue, but for now, I'm not sure this makes sense. There seem to be no product fields that are also hashes, so queries like `image.src` does not work. There are fields that are arrays of hashes (ex. images is an array of image objects), but the format doesn't seem to work for them anyway. That's all to say that there is no good way to test this directly.